### PR TITLE
[WIP] Experimental color log

### PR DESCRIPTION
**Pull request by nuclearpinguin**
Screenshot 2019-07-26 at 09 00 10

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by nuclearpinguin** [comment]
I am still thinking about pretty printing argument if it is a dictionary. The log will be easier to read but also longer...

**Comment by potiuk** [comment]
I think it's better to keep one-line per log entry. There are some really nice tools in IDEs to look at search  logs - for example ideolog: https://plugins.jetbrains.com/plugin/9746-ideolog. Also day before yesterday release of IntelliJ introduced capability of handling log files of arbitrary size easily - a lot of those tools rely on one log/lin.

"
IntelliJ IDEA works with large files smoothly
You can now easily open files larger than 25MB in IntelliJ IDEA and search through them without tedious waiting. This is a very nice little improvement for those of you who work with the log files."

**Comment by nuclearpinguin** [comment]
Regarding the log file, I think the colors should be only for console handler, do you agree?

**Comment by potiuk** [comment]
By default yes. I think you should be able to choose (but I guess you can using airflow configuration)

**Comment by nuclearpinguin** [comment]
@mik-laj any thoughts before I open PR to apache? I suppose there will be further discussion 😳

**Comment by mik-laj** [comment]
Yes. I have alternative simpler implementation

**Comment by mik-laj** [comment]
Today I will do a review.

File: airflow/utils/log/colored_log.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+        :param record: An log record.
+        :type record: logging.LogRecord
+        :return: Colored LogRecord.
+        """
+        record.filename = colored(record.filename, FILENAME_COLOR)
+
+        # Color level info
+        if record.levelname in ("ERROR", "DEBUG", "CRITICAL"):
+            record.msg = colored(record.msg, ERROR)
```
Comment (mik-laj) [inline airflow/utils/log/colored_log.py]: I think that coloring the entire message may make reading difficult.

Comment (nuclearpinguin) [reply airflow/utils/log/colored_log.py]: I was not sure about this. In some logs you can find red errors. I think we should limit coloring to (time, path, level) like it's done in docker. Additionally format the arguments.

**Pull request closed without merging**